### DISCUSSION TITLE
Validasi Cluster

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#768] [Q&A] [answered]
Dear rekan2 Algoritma,

Saya sedang membuat model clustering dengan menggunakan metode Hierarchical Clustering, dimana data yang saya gunakan jumlah observasinya  sebanyak 26,000 dengan 13 variabel.Kemudian saya juga sudah menghitung (dis)similarity antar cluster dengan menggunakan pendekatan linkage methode dengan hasil dari nilai korelasinya adalah sebagai berikut:
<img width="218" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/f5a34187-bab3-409a-85d4-8c8343b2f5c9">

Namun saya mendapat kendala pada saat akan melakukan validasi cluster, dimana komputasinya cukup besar dan memakan waktu yang cukup lama.
<img width="327" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/c61c8fc9-7cc6-4e77-a3de-33cb152f6451">

Untuk kasus ini, apakah ada cara untuk mempercepat proses komputasinya agar tidak terlalu lama. Saya sudah coba tes, jalankan syntax tersebut, sudah berjalan selama 4 jam namun belum selesai.

Tambahan: Jika di console saya ketik y seperti gambar dibawah apakah sudah jalan ya
<img width="238" alt="image" src="https://github.com/teamalgoritma/community/assets/103427989/f7d9007f-5726-4bd8-b0d5-1098e67983a2">

Mohon sarannya,

Best regards,
JT

**Comment by ranydwi** [comment]
Dear @joejnt  ,

Pertanyaan Anda sudah kami baca dan sedang kami usahakan untuk solusinya. Harap menunggu, terima kasih :)

**Comment by divakartika** [comment, marked as answer]
Dear @joejnt ,
Terkait waktu training yang lama, saat ini kami belum menemukan solusi terkait hal tersebut. Dari hasil pencarian kami, clValid memang cocok untuk data dengan ukuran hingga 600 baris, hal ini ditunjukkan dari parameter `maxitems`

> `maxitems = 600` The maximum number of items (rows in matrix) which can be clustered. ([rdocumentation](https://www.rdocumentation.org/packages/clValid/versions/0.7/topics/clValid))

Parameter tersebut juga yang menyebabkan adanya pesan konfirmasi `The number of items to be clustered is larger than 'maxitems'. The memory and time required may be excessive, do you wish to continue? (y to continue, any other character to exit)`. Jika diketikkan `y` lalu `enter` maka training akan berjalan. 

Untuk mencegah pesan tersebut muncul, parameter `maxitems` dapat diisi dengan jumlah row data, dapat dituliskan `maxitems = nrow(obj)`.

Referensi:
- [RDocumentation: clValid](https://www.rdocumentation.org/packages/clValid/versions/0.7/topics/clValid)
- [R clValid function Error for huge dataset](https://stackoverflow.com/questions/51211965/r-clvalid-function-error-for-huge-dataset)
- [library(clValid) - The number of items to be clustered is larger than 'maxitems'](https://stackoverflow.com/questions/44704151/libraryclvalid-the-number-of-items-to-be-clustered-is-larger-than-maxitems)

Semoga membantu dan menjawab pertanyaan Anda :)

Best regards,
Team Algoritma

**Reply by joejnt** [reply]
Dear @divakartika 

Terima kasih untuk infonya, untuk data dengan obs lebih dari 600 apakah masih bisa menggunakan library clValid ya. Kemudian apakah ada library selain dari clValid yang dapat digunakan untuk memvalidasi dengan obs data yang lebih besar.

Untuk yang ini ".....Jika diketikkan y lalu enter maka training akan berjalan." ngetiknya di console ya

Best regards,
Joe Takapente

**Reply by divakartika** [reply]
Dear @joejnt ,

Berdasarkan [referensi ini](https://stackoverflow.com/questions/51211965/r-clvalid-function-error-for-huge-dataset), `clValid` dapat memproses data hingga 65536 baris. Sebagai alternatif, dapat digunakan fungsi `NbClust()` dari library `NbClust`.

Berikut referensi code terkait penggunaan fungsi `NbClust()`: https://fxjollois.github.io/cours-2016-2017/stat-prog-R-2.html

Semoga membantu dan menjawab pertanyaan Anda :)

Best regards,
Team Algoritma

**Reply by joejnt** [reply]
Dear @divakartika 

Well noted, terima kasih untuk saran dan informasinya.

Best regards,
JT
